contracts/SpokePool.sol

```diff
@@ -1112,8 +1132,10 @@ abstract contract SpokePool is
         // slither-disable-next-line timestamp
         if (getCurrentTime() - quoteTimestamp > depositQuoteTimeBuffer) revert InvalidQuoteTimestamp();
 
-        // Increment count of deposits so that deposit ID for this spoke pool is unique.
-        uint32 newDepositId = numberOfDeposits++;
+        // Add the last known deposit count to the depositor's ID to ensure that all deposit IDs are unique including
+        // all past deposits which used numberOfDeposits as a counter.
+        uint32 newDepositId = depositCount(depositor);
+        depositorDepositCount[depositor]++;
```

Comment:
why does the deposit ID need to be an incremental counter? and not be a hash like a transaction hash (eth_sendRawTransaction) or a message hash (userOp)

Reply:
It's useful for looking up old deposits if we can assume they always increment. We can implement a binary search on them for example (I'll link to the offchain code we run in production)

But after talking with @mrice32 im not so convinced this property is important to preserve

Reply:
But at a minimum I don't want to change the type of depositId emitted in the V3FundsDeposited event because otherwise the upgrade path would be very tricky